frontend/workflows/dynamodb/src/update-capacity.tsx

```diff
@@ -0,0 +1,254 @@
+import React from "react";
+import type { clutch as IClutch } from "@clutch-sh/api";
+import {
+  Button,
+  ButtonGroup,
+  CheckboxPanel,
+  client,
+  MetadataTable,
+  Resolver,
+  Table,
+  TableRow,
+  useWizardContext,
+} from "@clutch-sh/core";
+import { useDataLayout } from "@clutch-sh/data-layout";
+import type { WizardChild } from "@clutch-sh/wizard";
+import { Wizard, WizardStep } from "@clutch-sh/wizard";
+import styled from "@emotion/styled";
+import _ from "lodash";
+import { number } from "yup";
+
+import type { ResolverChild, WorkflowProps } from "./index";
+
+const Container = styled.div({
+  display: "flex",
+});
+
+const TableIdentifier: React.FC<ResolverChild> = ({ resolverType }) => {
+  const { onSubmit } = useWizardContext();
+  const resolvedResourceData = useDataLayout("resourceData");
+
+  const onResolve = ({ results }) => {
+    // Decide how to process results.
+    resolvedResourceData.assign(results[0]);
+    onSubmit();
+  };
+
+  return <Resolver type={resolverType} searchLimit={1} onResolve={onResolve} />;
+};
+
+const TableDetails: React.FC<WizardChild> = () => {
+  const { onSubmit, onBack } = useWizardContext();
+  const resourceData = useDataLayout("resourceData");
+  const capacityUpdates = useDataLayout("capacityUpdates");
+  const table = resourceData.displayValue() as IClutch.aws.dynamodb.v1.Table;
+
+  const handleTableCapacityChange = (key: string, value: string) => {
+    const newTableCapacity = { ...capacityUpdates.displayValue().table_throughput };
+    newTableCapacity[key] = value;
+    capacityUpdates.updateData("table_throughput", newTableCapacity);
+  };
+
+  const handleGsiCapacityChange = (key: string, value: string) => {
+    // big hack to retrieve the capacity type (read or write?) and
+    // the GSI name from a single event attribute [key]
+    // where key is formatted like "read,gsi-name"
+    // feature request to address this: https://github.com/lyft/clutch/issues/1739
+    const keys = key.split(",");
+    const capacityType = keys[0];
+    const gsiName = keys[1];
+
+    const gsiList = capacityUpdates.displayValue().gsi_updates
+      ? [...capacityUpdates.displayValue().gsi_updates]
+      : [];
+    const idx = gsiList.findIndex(
+      (gsi: { name: string; indexThroughput: {} }) => gsi.name === gsiName
+    );
+    if (idx > -1) {
+      // gsi already in the edits list
+      const gsi = gsiList[idx];
+      gsi.index_throughput = { ...gsi.index_throughput, [capacityType]: value };
+      capacityUpdates.updateData("gsi_updates", gsiList);
+    } else {
+      // copy over current capacities
+      const curr = table.globalSecondaryIndexes.find(gsi => gsi.name === gsiName);
```

Comment:
maybe this field should be a map from the server instead of repeated if we have to do this kind of searching for a unique name? or we could make it into a map at the beginning when ingesting the data.

Reply:
Mapped the GSIs to make searching for names easier 👍